### PR TITLE
Rwjs/animation stop

### PR DESCRIPTION
**Pull request by robsimmonds**
Code that introduces the animation scheme defined in the ICD.

### CODE REVIEW

File: Main.cc [hunk 1]
```diff
@@ -94,12 +96,19 @@ void OnMessage(uWS::WebSocket<uWS::SERVER>* ws, char* raw_message, size_t length
             OnMessageTask* tsk = nullptr;
 
             switch (head.type) {
+			    case CARTA::EventType::REGISTER_VIEWER: {
```
Comment (veggiesaurus) [inline Main.cc]: this section appears to use tabs instead of spaces

File: Main.cc [hunk 2]
```diff
@@ -143,23 +152,40 @@ void OnMessage(uWS::WebSocket<uWS::SERVER>* ws, char* raw_message, size_t length
                     session->StopAnimation(message.file_id(), message.end_frame());
                     break;
                 }
-                default: { tsk = new (tbb::task::allocate_root()) MultiMessageTask(session, head, event_length, event_buf); }
+	        case CARTA::EventType::ANIMATION_FLOW_CONTROL: {
```
Comment (veggiesaurus) [inline Main.cc]: also some mixed tab/space use here

File: AnimationObject.h [hunk 3]
```diff
@@ -26,25 +29,41 @@ class AnimationObject {
...
     int _wait_duration_ms;
-
-public:
-    AnimationObject(int file_id, CARTA::AnimationFrame& start_frame, CARTA::AnimationFrame& end_frame, CARTA::AnimationFrame& delta_frame,
-        int frame_interval, bool looping, bool reverse_at_end, CARTA::CompressionType compression_type, float compression_quality,
-        bool always_wait)
+    volatile int _file_open;
+    volatile bool _waiting_flow_event;
+    tbb::task * _waiting_task;
+ public:
```
Comment (confluence) [inline AnimationObject.h]: This shouldn't be indented with a space.

File: Session.cc [hunk 4]
```diff
@@ -121,12 +122,18 @@ void Session::ResetFileInfo(bool create) {
...
-void Session::OnRegisterViewer(const CARTA::RegisterViewer& message, uint32_t request_id) {
+void Session::OnRegisterViewer(const CARTA::RegisterViewer& message, uint16_t icd_version, uint32_t request_id) {
     auto session_id = message.session_id();
     bool success(false);
     std::string error;
     CARTA::SessionType type(CARTA::SessionType::NEW);
-    // check session id
+	
+	if (icd_version != carta::ICD_VERSION) {
+		fprintf(stderr,"%p : Existing due to wrong ICD version number. Expected %d, got %d.\n", this, carta::ICD_VERSION, icd_version);
```
Comment (confluence) [inline Session.cc]: Typo: existing -> exiting? (Also technically incorrect use of "due to" -- maybe just "Exiting: wrong ICD (...)" or "Wrong ICD (...). Exiting." ?)

**Pull request merged**